app/MMB/src/scenes/tickets/eTicketPdf/ETicketPdf.js

```diff
@@ -3,33 +3,28 @@
 import * as React from 'react';
 import { GeneralError } from '@kiwicom/mobile-shared';
 import { Translation } from '@kiwicom/mobile-localization';
-import PdfViewer from '@kiwicom/mobile-pdf';
+import { PdfDownloader } from '@kiwicom/mobile-pdf';
 
 import BookingDetailContext from '../../../context/BookingDetailContext';
 
 type PropsWithContext = {|
   ...Props,
-  +setETicketPath: (path: string) => void,
+  +bookingId: string,
 |};
 
-class ETicketPdf extends React.Component<PropsWithContext> {
-  onPdfLoaded = (pages: number, path: string) => {
-    this.props.setETicketPath(path);
-  };
+const ETicketPdf = ({ ticketUrl, bookingId }: PropsWithContext) => {
+  if (!ticketUrl) {
+    return (
+      <GeneralError
+        errorMessage={<Translation id="mmb.tickets.not_available" />}
+      />
+    );
+  }
 
-  render = () => {
-    const { ticketUrl } = this.props;
-    if (!ticketUrl) {
-      return (
-        <GeneralError
-          errorMessage={<Translation id="mmb.tickets.not_available" />}
-        />
-      );
-    }
-
-    return <PdfViewer uri={ticketUrl} onLoad={this.onPdfLoaded} />;
-  };
-}
+  return (
+    <PdfDownloader fileName={`/eTickets/${bookingId}.pdf`} url={ticketUrl} />
```

Comment:
I would expect here only the `PdfViewer` here because this is what we want. This component should care of downloading these PDFs. Also, I am not sure about the filename change. Shouldn't we keep the original PDF name?

Reply:
My idea was to do the logic about showing loader while downloading, keeping the url contained in one file, instead of having to do this

```

Reply:
Also about keeping the original name, I think it makes sense to store it by `/tickets/<bid>.pdf` how else would we easily find the file, if we not want to put it in a context?

The existing name seems kind of random

Reply:
I will remind you this invention after we'll start generating multiple e-tickets for one BID... :)

Reply:
Are there multiple e-tickes for one BID?

Reply:
No, not yet.

Reply:
Will it ever be? Will it ever make sense?

If you have any better means of storing it, which makes it easy to find, I am down for implementing it

Reply:
I don't know. Think about the future. It may happen. Can we fetch the names from the API where needed?

Reply:
We would need to parse the url, get the name, we could probably do this on graphql

Reply:
OK, forget it for now.